service/history/workflow/update/update.go

```diff
@@ -253,12 +255,13 @@ func (u *Update) WaitLifecycleStage(
 
 		if ctx.Err() != nil {
 			// Handle root context deadline expiry as normal error which is returned to the caller.
+			metrics.WorkflowExecutionUpdateClientTimeout.With(u.instrumentation.metrics).Record(1)
 			return nil, ctx.Err()
 		}
+	}
 
-		if stCtx.Err() != nil {
-			return statusAdmitted(), nil
-		}
+	if stCtx.Err() != nil {
+		metrics.WorkflowExecutionUpdateServerTimeout.With(u.instrumentation.metrics).Record(1)
 	}
```

Comment:
I removed this since then it will fallthrough to the same line at the bottom.

Reply:
Then you need to update comment bellow adding `or soft timeout has expired`. I removed it because you asked me for it :-)